src/modules/75-cf/components/CreateEnvironmentDialog/__tests__/EnvironmentDialog.test.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import { render, RenderResult, screen, waitFor, fireEvent } from '@testing-library/react'
+import React from 'react'
+import { TestWrapper } from '@common/utils/testUtils'
+import * as useFeaturesMock from '@common/hooks/useFeatures'
+import * as usePlanEnforcementMock from '@cf/hooks/usePlanEnforcement'
+import { FeatureIdentifier } from 'framework/featureStore/FeatureIdentifier'
+
+import EnvironmentDialog, { EnvironmentDialogProps } from '../EnvironmentDialog'
+
+const renderComponent = (props: Partial<EnvironmentDialogProps> = {}): RenderResult => {
+  return render(
+    <TestWrapper
+      path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/feature-flags"
+      pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
+    >
+      <EnvironmentDialog onCreate={jest.fn()} {...props} />
+    </TestWrapper>
+  )
+}
+
+describe('CreateEnvironmentButton', () => {
+  beforeEach(() =>
+    jest.spyOn(usePlanEnforcementMock, 'default').mockReturnValue({ isPlanEnforcementEnabled: true, isFreePlan: false })
+  )
+
+  test('it should display plan enforcement tooltip when limits reached', async () => {
+    jest
+      .spyOn(useFeaturesMock, 'useGetFirstDisabledFeature')
+      .mockReturnValue({ featureEnabled: false, disabledFeatureName: FeatureIdentifier.MAUS })
+
+    renderComponent()
+
+    fireEvent.mouseOver(screen.getByText('+ environment'))
```

Comment:
So we're targeting the mouse over based on the text of the button, but you also just added a `data-testid` to the element, can we just use that instead?
```suggestion
    fireEvent.mouseOver(screen.getByTestId('create-environment-button'))
```

Reply:
react-testing-library recommends to "test in the way your software is used". In this case the `getByText` more resembles what the user would look for (ie, the button with `+ environment)

Naturally this isn't always feasible thats why getByTestId exists

Reply:
good to know, thanks!

Reply:
`getByRole` is what you really want. The user isn't looking to click on _"+ environment"_, they're looking to click on a **button** that says _"+ environment"_.

Reply:
Personally I'd rip out `getByTestId` and setting the `data-testid` in the component and just use `getByRole`.